README.md

```diff
@@ -1,58 +1,2501 @@
+kubectl get secret all-icr-io -o yaml | sed 's/namespace: .*/namespace: kube-system/' | kubectl apply -f -
+```
+> Note: If the `all-icr-io` secret does not exist in the default namespace, follow the instructions in this document to apply it: https://cloud.ibm.com/docs/containers?topic=containers-registry#imagePullSecret_migrate_api_key
 
-* Ensure the pods are Running
-`kubectl get po -n kube-system | grep ibm-cloud-ingress`
+6. Apply the deployment.
+```
+kubectl apply -f ./sample_deploy
+```
+
+7. Ensure the Ingress controller pods are in a `Running` state.
+```
+kubectl get po -n kube-system | grep ibm-cloud-ingress
+```
  - If not, check the events for errors:
-   `kubectl describe deploy -n kube-system ibm-cloud-ingress`
-
-### Register DNS domain
-* Get your service IP(Classic) or LB Domain(VPC)
-`kubectl get svc -n kube-system | grep ibm-cloud-ingress`
-  - note: The IP or domain is the 4th column
-
-* Create a DNS domain for your ingress
-  - for classic:
-`ibmcloud ks nlb-dns create classic -c <cluster> --ip <service ip address>`
-  - for VPC:
-`ibmcloud ks nlb-dns create vpc-gen2 -c <cluster> --lb-host <lb hostname>`
\ No newline at end of file
+   ```
+   kubectl describe deploy -n kube-system ibm-cloud-ingress
+   ```
+
+TODO: is this always going to be a public ingress controller w a public ip address (or hostname)? or can they specify private somehow?
+
+## Step 3: Registering a DNS subdomain
+Register the service that exposes the Ingress controller with an IBM-provided DNS subdomain or a custom domain.
+
+### IBM-provided domain
+1. Get the ingress controller service's IP address (classic clusters) or VPC load balancer hostname (VPC clusters). The IP or hostname is the 4th column of the output.
+```
+kubectl get svc -n kube-system | grep ibm-cloud-ingress
+```
+
+2. Create a DNS subdomain for your Ingress controller. When you run the following command, an INM-provided subdomain is automatically generated and registered with the ingress controller. A TLS secret for the domain is automatically generated in the namespace that you can optionally specify; otherwise, the secret is created in the `default` namespace. The secret name follows a truncated format of the subdomain, such as `mycluster-a1b2cdef345678g9hi012j3kl4567890-0003`.
+  - Classic:
+    ```
+    ibmcloud ks nlb-dns create classic -c <cluster> --ip <service ip address> [--namespace <namespace>]
+    ```
+  - VPC:
+  ```
+  ibmcloud ks nlb-dns create vpc-gen2 -c <cluster> --lb-host <lb hostname> [--namespace <namespace>]
+  ```
+TODO: if they can create a private controller, can include  --type private for vpc or set up a private dns resolution svc for classic ("Classic clusters with worker nodes that are connected to [a private VLAN only](cloud.ibm.com/docs/containers?topic=containers-cs_network_planning#plan_private_vlan): Configure a [DNS service that is available on the private network](https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/).")
```

Comment:
yeah, it may be worth noting that private can be done, but modifications will need to be done by the user to do so.